lib/devcycle-ruby-server-sdk/api/dev_cycle_provider.rb

```diff
@@ -32,7 +32,18 @@ def fetch_number_value(flag_key:, default_value:, evaluation_context: nil)
     end
 
     def fetch_integer_value(flag_key:, default_value:, evaluation_context: nil)
-      @client.variable(Provider.user_from_openfeature_context(evaluation_context), flag_key, default_value)
+      variable = @client.variable(Provider.user_from_openfeature_context(evaluation_context), flag_key, default_value)
+
+      eval_reason = variable.isDefaulted ? OpenFeature::SDK::Provider::Reason::DEFAULT : OpenFeature::SDK::Provider::Reason::TARGETING_MATCH
+
+      if variable.eval && variable.eval.is_a?(Hash) && variable.eval['reason'] == 'SPLIT'
+        eval_reason = OpenFeature::SDK::Provider::Reason::SPLIT
+      end
+
+      OpenFeature::SDK::Provider::ResolutionDetails.new(
```

Comment:
I'm confused by this, we are returning a `ResolutionDetails` here and a `DevCycle.Varaible` for the other types?